### ISSUE TITLE
adding parameters when inserting a method

### ISSUE THREAD

**Issue by gp187** [dsherret/ts-simple-ast#262]
Maybe I'm doing something wrong, but I think this isn't working properly. I'm trying to add a method to a function and it isn't setting the parameters, but when I do it in a 2 step process, it works. 

Input to 
```typescript
const meth = {
                name: 'function1', scope: Scope.Public,
                parameters: [{name: 'i', type: 'number'}],
                returnType: 'void',
                bodyText: 'console.log(i);'
};
myClass.addMethod(meth);
console.log(meth) // the output no longer has the `parameters` key
```
Result:
```typescript
function1(): void {
   console.log(i)
}
```

**parameters is gone**

Now, if after all of this I get the method and add parameters, it works

```typescript
const f = myClass.getInstanceMethod('function1');
f = f.addParameters([{name: 'i', type: 'number'}])
```
Now the function has the call parameters it should have had in the first place. 
Now this is where it gets **special**

```typescript
console.log(meth) // output as you'd expect
myClass.addMethod(meth);
console.log(meth) // the output no longer has the `parameters` key
```

I did a work around by passing `params` key with the same values and adding them after the method is added and works fine. 

Removing `parameters` is normal behaviour?

**Comment by dsherret** [comment]
Hey @gp187, what version are you using? I just tested this out and it's working for me. One issue is #263, which I just opened.

This probably somehow relates to some performance improvements I'm slowly working on... it's doing things in two ways right now. Also, the structure being modified after being passed in is my bad. I should have been editing a copy. I'll fix that.

**Comment by dsherret** [comment]
I'm not sure why I referenced this in that previous commit. That was wrong... anyway, this should definitely not be happening in my latest commits for #224, but that will be released in (10.2.0 or 10.3.0)
